packages/backend/src/services/RenameService/RenameService.ts

```diff
@@ -0,0 +1,421 @@
+import { subject } from '@casl/ability';
+import {
+    AlreadyExistsError,
+    ApiRenameBody,
+    ApiRenameResponse,
+    ChartSummary,
+    DashboardDAO,
+    Explore,
+    ForbiddenError,
+    getDeepestPaths,
+    getErrorMessage,
+    getFieldRef,
+    getItemId,
+    isDashboardChartTileType,
+    isExploreError,
+    isSubPath,
+    NotFoundError,
+    ParameterError,
+    PromotedChart as PromotedChangeChart,
+    PromotedSpace,
+    PromotionAction,
+    PromotionChanges,
+    RenameResourcesPayload,
+    RenameType,
+    RequestMethod,
+    SavedChartDAO,
+    SCHEDULER_TASKS,
+    SchedulerAndTargets,
+    SessionUser,
+    Space,
+    SpaceShare,
+    SpaceSummary,
+    UnexpectedServerError,
+} from '@lightdash/common';
+import { cloneDeep } from 'lodash';
+
+import { LightdashAnalytics } from '../../analytics/LightdashAnalytics';
+import { LightdashConfig } from '../../config/parseConfig';
+import Logger from '../../logging/logger';
+import { DashboardModel } from '../../models/DashboardModel/DashboardModel';
+import { ProjectModel } from '../../models/ProjectModel/ProjectModel';
+import { SavedChartModel } from '../../models/SavedChartModel';
+import { SchedulerModel } from '../../models/SchedulerModel';
+import { SpaceModel } from '../../models/SpaceModel';
+import { SchedulerClient } from '../../scheduler/SchedulerClient';
+import { BaseService } from '../BaseService';
+import { renameAlert, renameDashboard, renameSavedChart } from './rename';
+
+type RenameServiceArguments = {
+    lightdashConfig: LightdashConfig;
+    analytics: LightdashAnalytics;
+    projectModel: ProjectModel;
+    savedChartModel: SavedChartModel;
+    dashboardModel: DashboardModel;
+    schedulerClient: SchedulerClient;
+    schedulerModel: SchedulerModel;
+};
+
+export class RenameService extends BaseService {
```

Comment:
New service for this feature, at first, I added these methods to ValidationService and PromotionService, but they have different dependencies and extra private methdos, so I decided to create a  new one